app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java

```diff
@@ -0,0 +1,205 @@
+     * (g3) - modified by Rossi Leonardo - modified by Andrea Ton (barcode)
+     */
+    @SuppressLint("StaticFieldLeak")
+    private class AsyncLoad extends AsyncTask<Bitmap, Void, String> {
+
+        private ProgressDialog progressDialog;
+        private TextView resultTextView;
+        private String progressMessage;
+
+        AsyncLoad(TextView view, String progressMessage) {
+            this.resultTextView = view;
+            this.progressMessage = progressMessage;
+        }
+
+        @Override
+        protected String doInBackground(Bitmap... bitmaps) {
+            TextExtractor ocr = new TextExtractor();
+            String textRecognized = "";
+            String barcodeRecognized = "";
+
+            if(lastPhoto != null) {
+                //invoke a barcodeRecognizer choosing the api type
+                Barcode barcodeRecognizer = barcodeRecognizer(BarcodeRecognizer.API.mlkit);
+                //get the String of the barcode from the bitmap image
+                barcodeRecognized = barcodeRecognizer.decodeBarcode(lastPhoto);
+                //check if any barcode is found
+                if(barcodeRecognized.equals(""))
+                {
+                    //if no barcode was found than try to get the text ocr
+                    textRecognized = ocr.getTextFromImg(lastPhoto);
+
+                    if(textRecognized.equals(""))
+                    {
+                    //if nothing was found than return a No text found message
+                    textRecognized = getString(R.string.no_text_found);
+                    final String finalTextRecognized = textRecognized;
+
+                    runOnUiThread(() -> {
+                            mOCRTextView.setText(finalTextRecognized);
+                        });
+                    }
+                    else {
+                        //if text is found, return it
+                        final String finalTextRecognized = textRecognized;
+
+                        runOnUiThread(() -> {
+                            mOCRTextView.setText(finalTextRecognized);
+                        });
+                    }
+                }
+                else {
+                    //if barcode id found return the associated number, as a String,
+                    //with a fixed "Barcode: " string to check the result is not from the ocr
+                    final String finalTextRecognized ="Barcode: " + barcodeRecognized;
+
+                    runOnUiThread(() -> {
+                        mOCRTextView.setText(finalTextRecognized);
+                    });
+                }
+            }
+            else {
+                Log.e("NOT_FOUND", "photo not found");
+            }
+            //using two different variables and one more if condition to avoid string conflict
+            if(barcodeRecognized.equals("")) {
+                return textRecognized;
+            }
+            else {
+                return barcodeRecognized;
+            }
+
```

Comment:
just give a different name to the variable "s" something more detailed

Reply:
of course, thank you